File: app/javascript/services/CollegeApiService.js
import fetch from 'isomorphic-unfetch'

const ENDPOINT = 'http://localhost:3000/colleges'

export default class CollegeApiService {
  static async getColleges({ search }) {
    const url = `${ENDPOINT}?search=${search}`

    try {
      const response = await fetch(url)
      return await response.json()
    } catch (error) {
      return { error }
    }
  }
}

File: app/services/college_api_service.rb
class CollegeApiService
  ENDPOINT = 'https://api.data.gov/ed/collegescorecard/v1/schools'
  FIELDS = 'id,school.name,location'

  def self.execute(search)
    url = "#{ENDPOINT}?api_key=#{ENV['COLLEGE_SCORECARD_API_KEY']}"
    url += "&fields=#{FIELDS}&per_page=100&school.name=#{search}"

    response = HTTP.get(url)
    JSON.parse(response.body)['results']
  end
end
